src/reducers/groupReducer.js

```diff
@@ -0,0 +1,98 @@
+import { DEFAULT_DATA } from "../appConfig";
+import * as actions from "../actions";
+
+const todoListFilterFunction = (todoGroups, todoListFilter, payload) => {
+   return todoGroups.map((groupItem) => {
+      const todoList = groupItem.todoList.map((todoItem) => todoListFilter(todoItem, payload));
+      return { ...groupItem, todoList };
+   });
+};
+
+const filterTodoByDone = (todoListItem, completed) => {
+   if (completed === true) {
+      return { ...todoListItem, sortByDone: todoListItem.done };
+   } else {
+      return { ...todoListItem, sortByDone: null };
+   }
+};
+
+const filterTodoByTitle = (todoListItem, searchTitle) => {
+   if (searchTitle.length > 0) {
+      return { ...todoListItem, sortByTitle: todoListItem.title.startsWith(searchTitle) };
+   } else {
+      return {
+         ...todoListItem,
+         sortByTitle: null,
+      };
+   }
+};
+
+const groupListFilterFunction = (filterArray, callFilter, payload) => {
+   return filterArray.map((group) => callFilter(group, payload));
+};
+
+const searchGroupByColor = (group, searchColor) => {
+   if (searchColor === "white") {
+      return { ...group, sortByColor: true };
```

Comment:
Take a look on "filterTodoByTitle" function. You setting "sortByTitle = null" when filter is disabled. Please do the same for "searchGroupByColor" and "searchGroupByCreated". (set filter marker to null if filter disabled)